modules/ww-bcftools/README.md

```diff
@@ -0,0 +1,197 @@
+}
+
+workflow my_variant_calling_pipeline {
+  input {
+    Array[SampleInfo] samples
+    File reference_fasta
+    File reference_fasta_index
+    File? regions_bed
+  }
+  
+  scatter (sample in samples) {
+    call bcftools_tasks.mpileup_call {
+      input:
+        sample_data = sample,
+        reference_fasta = reference_fasta,
+        reference_fasta_index = reference_fasta_index,
+        regions_bed = regions_bed
+    }
+  }
+  
+  output {
+    Array[File] variant_vcfs = mpileup_call.output_vcf
+  }
+}
+```
+
+### Integration Examples
+
+This module integrates well with other WILDS components:
+- **ww-star**: Use aligned BAM files from STAR for variant calling
+- **ww-sra**: Download data, align, then call variants
+- **Custom workflows**: Combine with GATK or other variant callers for consensus calling
+
+## Testing the Module
+
+The module includes a demonstration workflow with comprehensive testing:
+
+```bash
+# Using Cromwell
+java -jar cromwell.jar run ww-bcftools.wdl --inputs inputs.json --options options.json
+
+# Using miniWDL
+miniwdl run ww-bcftools.wdl -i inputs.json
+
+# Using Sprocket
+sprocket run ww-bcftools.wdl inputs.json
+```
+
+### Test Input Format
+
+```json
+{
+  "bcftools_example.samples": [
+    {
+      "name": "sample1",
+      "bam": "/path/to/sample1.bam",
+      "bai": "/path/to/sample1.bam.bai"
+    }
+  ],
+  "bcftools_example.reference_genome": {
+    "name": "hg38",
+    "fasta": "/path/to/genome.fasta",
+    "fasta_index": "/path/to/genome.fasta.fai"
+  },
+  "bcftools_example.regions_bed": "/path/to/regions.bed",
+  "bcftools_example.max_depth": 10000,
+  "bcftools_example.memory_gb": 8,
+  "bcftools_example.cpu_cores": 2
+}
+```
+
+## Configuration Guidelines
+
+### Resource Allocation
+
+The module supports flexible resource configuration:
+- **Memory**: 4-8GB recommended for most applications
+- **CPUs**: 1-2 cores typically sufficient for bcftools operations
+- **Storage**: Sufficient space for input BAMs and output VCFs
+
+### Variant Calling Parameters
+
+- `max_depth`: Adjust based on expected coverage depth
+- `max_idepth`: Set higher for high-coverage applications
+- `annotate_format`: Customize FORMAT fields in output VCF
+- `regions_bed`: Use for targeted sequencing applications
+
+## Requirements
+
+- WDL-compatible workflow executor (Cromwell, miniWDL, Sprocket, etc.)
+- Docker/Apptainer support
+- Input BAM files must be sorted and indexed
+- Reference genome with FASTA index
+
+## Features
+
+- **Variant calling**: Configurable mpileup/call parameters
+- **Targeted regions**: Support for BED file region specification
+- **Validation**: Built-in output validation and statistics
+- **Scalable**: Configurable resource allocation
+- **Compatible**: Works with multiple WDL executors
+
+## Advanced Usage
+
+The module supports various bcftools mpileup/call parameters for fine-tuning variant calling:
+
+```wdl
+call bcftools_tasks.mpileup_call {
+  input:
+    sample_data = my_sample,
```

Comment:
```suggestion
    sample_data = my_sample,
    regions_bed = my_bedfile,
```

Consider adding this to show that users can specify their own regions of interest. Or not because it's an optional input.